File: src/main/java/DayOneCaptcha.java
public class DayOneCaptcha {
    int captchaSum(String input) {
        int sum = 0;
        input = input + input.charAt(0);
        for (int i = 0; i < input.length() - 1; i++) {
            char next = input.charAt(i + 1);
            char current = input.charAt(i);
            if (current == next) {
                sum += Character.getNumericValue(current);
            }
        }
        return sum;
    }

    int captchaHalfwaySum(String input){
        int sum = 0;
        for (int i = 0; i < input.length()/2; i++) {
            char next = input.charAt(i + input.length()/2);
            char current = input.charAt(i);
            if (current == next) {
                sum += Character.getNumericValue(current);
            }
        }
        sum = sum * 2;
        return sum;
    }
}

File: src/main/java/DayOneCaptcha.kt
class DayOneCaptcha {
    fun captchaSum(input: String): Int {
        var workingString = input
        var sum = 0
        workingString += workingString[0]
        for (i in 0 until workingString.length - 1) {
            val next = workingString[i + 1]
            val current = workingString[i]
            if (current == next) {
                sum += Character.getNumericValue(current)
            }
        }
        return sum
    }

    fun captchaHalfwaySum(input: String): Int {
        var sum = 0
        for (i in 0 until input.length / 2) {
            val next = input[i + input.length / 2]
            val current = input[i]
            if (current == next) {
                sum += Character.getNumericValue(current)
            }
        }
        sum *= 2
        return sum
    }
}
